src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteSpec.groovy

```diff
@@ -156,10 +156,34 @@ class AutoRerouteSpec extends HealthCheckSpecification {
         errorDetails.errorDescription.contains("Not enough bandwidth or no path found")
 
         and: "Flow remains DEGRADED and on the same path"
-        wait(rerouteDelay + WAIT_OFFSET) { //2 more reroute attempts (reroute + retry)
-            assert northbound.getFlowHistory(flow.flowId).size() == history.size() + 2
+        wait(rerouteDelay + WAIT_OFFSET) {
+            assert northbound.getFlowHistory(flow.flowId).findAll {
+                it.action == REROUTE_ACTION && it.details == "Reason: initiated via Northbound"
+            }.size() == 2 //reroute + retry
             assert northboundV2.getFlowStatus(flow.flowId).status == FlowState.DEGRADED
         }
+        PathHelper.convert(northbound.getFlowPath(flow.flowId)) == pathAfterReroute1
+        pathHelper.getInvolvedIsls(pathAfterReroute1).each {
+            assert northbound.getLink(it).availableBandwidth == flow.maximumBandwidth - 1 }
+
+        when: "Trigger auto reroute by blinking not involved(in flow path) isl"
+        def islToBlink = topology.islsForActiveSwitches.find {
+            [it.srcSwitch.dpId, it.dstSwitch.dpId].intersect([flow.source.switchId, flow.destination.switchId]).empty
+        }
+        antiflap.portDown(islToBlink.srcSwitch.dpId, islToBlink.srcPort)
+        def islToBlinkIsDown = true
+        antiflap.portUp(islToBlink.srcSwitch.dpId, islToBlink.srcPort)
+        islToBlinkIsDown = false
+
+        then: "System tries to reroute the DEGRADED flow"
+        and: "Flow remains DEGRADED and on the same path"
+        wait(rerouteDelay + WAIT_OFFSET) {
+            assert northbound.getFlowHistory(flow.flowId).findAll {
+                it.action == REROUTE_ACTION && it.details.contains("status become ACTIVE")
+            }.size() == 2 //reroute + retry
```

Comment:
This is not dealing with reroutes coming from parallel tests/islands. may become unstable

Reply:
Same for the above history search with `"Reason: initiated via Northbound"`. The retry may merge the reroute details with one of the 'become Active' reasons from parallel topologies

Reply:
moved to isolated spec